### PR TITLE
[T-12] Edit Account Details

### PR DESCRIPTION
**Pull request by absirgue**
Hello reviewers,

I hope you're having a great day!

Here is my PR for the edit account page.

It includes:
- the page itself
- edits of the auth.dart class to accomodate that change
- Unit and Widget tests for this task (the model used for testing - especially widget testing- can now be reused for other tasks as it allows to test any page without worrying about the navigation that used to bother us)

In the process, I also edited a few rules so them and their tests are also on this branch.

Let me know what you think, happy to make any change required and answer to any review !

Thanks,

### CODE REVIEW

File: test/screens/admin/edit_account_test.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+void main() {
+  testWidgets("Edit account page has all components",
+      (WidgetTester tester) async {
+    mockNetworkImagesFor(() async {
+      Firebase.initializeApp();
+      await tester.pumpWidget(const MaterialApp(home: EditAccountPage()));
+
+      expect(find.byKey(const ValueKey("edit_email_submit")), findsWidgets);
+      expect(find.byKey(const ValueKey("edit_email_prompt")), findsWidgets);
+      // expect(find.byKey(const ValueKey("edit_email_field")), findsWidgets);
```
Comment (skellyhoe) [inline test/screens/admin/edit_account_test.dart]: remove commented out code

**Review by skellyhoe — CHANGES_REQUESTED** [review]
Nice work!!

Here's some comments to consider:

- Write general comments for the loading indicator explaining where it's used and what it does.
- Write general comments on the edit_account page and write some comments for the methods used.
- Split the "Edit your email" and "Edit your password" in the edit_account page and make them 2 widgets to be called from the edit_account page.
- Add some padding to the texts and buttons in the edit_account page to make the look cleaner.
- Remove commented out code highlighted below.

File: lib/pages/authenticate/edit_account.dart [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+                              height: 60,
+                              width: 250,
+                              child: ElevatedButton(
+                                key: Key('edit_email_submit'),
+                                style: ElevatedButton.styleFrom(
+                                  textStyle: TextStyle(
+                                    fontSize: 20,
+                                    fontWeight: FontWeight.bold,
+                                  ),
+                                  foregroundColor: Colors.white,
```
Comment (skellyhoe) [inline lib/pages/authenticate/edit_account.dart]: remove every hard coded color for the buttons

File: lib/widgets/edit_password_section.dart [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+              height: 60,
+              width: 250,
+              child: ElevatedButton(
+                key: Key('edit_password_submit'),
+                style: ElevatedButton.styleFrom(
+                  textStyle: TextStyle(
+                    fontSize: 20,
+                    fontWeight: FontWeight.bold,
+                  ),
+                  foregroundColor: Colors.white,
```
Comment (skellyhoe) [inline lib/widgets/edit_password_section.dart]: hard coded color

File: lib/widgets/edit_password_section.dart [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                  backgroundColor: Colors.teal[400],
```
Comment (skellyhoe) [inline lib/widgets/edit_password_section.dart]: hard coded color

File: lib/widgets/edit_email_section.dart [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                        height: 60,
+                        width: 250,
+                        child: ElevatedButton(
+                          key: Key('edit_email_submit'),
+                          style: ElevatedButton.styleFrom(
+                            textStyle: TextStyle(
+                              fontSize: 20,
+                              fontWeight: FontWeight.bold,
+                            ),
+                            foregroundColor: Colors.white,
```
Comment (skellyhoe) [inline lib/widgets/edit_email_section.dart]: hard coded color

File: lib/widgets/edit_email_section.dart [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                            backgroundColor: Colors.teal[400],
```
Comment (skellyhoe) [inline lib/widgets/edit_email_section.dart]: hard coded color

**Review by skellyhoe — CHANGES_REQUESTED** [review]
All good. Just remove the hard coded colors and it'll be perfect.

File: lib/pages/authenticate/edit_account.dart [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    _currentPasswordController.dispose();
+    _newPasswordController.dispose();
+    _confirmPasswordController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.grey[200],
```
Comment (skellyhoe) [inline lib/pages/authenticate/edit_account.dart]: hard coded color

**Review by skellyhoe — APPROVED** [review]
Thank you habibi.

File: lib/pages/authenticate/edit_account.dart [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+class EditAccountPage extends StatefulWidget {
+  const EditAccountPage({
+    super.key,
+  });
+
+  @override
+  State<EditAccountPage> createState() => EditAccountPageState();
+}
+
+class EditAccountPageState extends State<EditAccountPage> {
```
Comment (Ali-Alkhars) [inline lib/pages/authenticate/edit_account.dart:23]: Add a general comment here explaining the functionality of this class

File: lib/services/auth.dart [hunk 9]
```diff
@@ -0,0 +1,114 @@
+
+import 'package:firebase_auth/firebase_auth.dart';
+
+class Auth {
```
Comment (Ali-Alkhars) [inline lib/services/auth.dart:4]: General comment here too

**Review by Ali-Alkhars — CHANGES_REQUESTED** [review]
Great work habibi!

Please do the following:
- Add missing comments pointed out below
- Show a more accurate error message for when changing the email to a pre-existing email

**Review by AryamanDubey28 — APPROVED** [review]
Great work on the rules, testing and Edit account UI!

Just get those integration tests passing and alls good

**Review by Ali-Alkhars — APPROVED** [review]
Well done! Thanks for making the requested changes

**Pull request merged**